modules/linear_boltzmann_solvers/a_lbs_solver/lbs_solver.cc

```diff
@@ -2288,282 +2437,237 @@ LBSSolver::MakeSourceMomentsFromPhi()
-  {
-    Chi::log.LogAllWarning() << __FUNCTION__ << "Failed to open " << file_name;
-    return;
-  }
-
-  // Get relevant items
-  auto NODES_ONLY = UnknownManager::GetUnitaryUnknownManager();
-  auto& sdm = discretization_;
-  uint64_t num_local_nodes = discretization_->GetNumLocalDOFs(NODES_ONLY);
-  uint64_t num_moments_t = static_cast<uint64_t>(num_moments_);
-  uint64_t num_groups_t = static_cast<uint64_t>(num_groups_);
-  uint64_t num_local_dofs = discretization_->GetNumLocalDOFs(flux_moments_uk_man_);
-  uint64_t num_local_cells = grid_ptr_->local_cells.size();
+  // Read the header
+  const int num_bytes = 500;
+  char header_bytes[num_bytes];
+  header_bytes[num_bytes - 1] = '\0';
+  file.read(header_bytes, num_bytes - 1);
 
+  // Read the macro info
+  uint64_t file_num_local_cells;
   uint64_t file_num_local_nodes;
   uint64_t file_num_moments;
   uint64_t file_num_groups;
-  uint64_t file_num_local_dofs;
-  uint64_t file_num_local_cells;
-
-  flux_moments.assign(num_local_dofs, 0.0);
-
-  // Read header
-  char header_bytes[500];
-  header_bytes[499] = '\0';
-  file.read(header_bytes, 499);
 
+  file.read((char*)&file_num_local_cells, sizeof(uint64_t));
   file.read((char*)&file_num_local_nodes, sizeof(uint64_t));
   file.read((char*)&file_num_moments, sizeof(uint64_t));
   file.read((char*)&file_num_groups, sizeof(uint64_t));
-  file.read((char*)&file_num_local_dofs, sizeof(uint64_t));
-  file.read((char*)&file_num_local_cells, sizeof(uint64_t));
 
-  // Check compatibility
-  if (not single_file)
-    if (file_num_local_nodes != num_local_nodes or file_num_moments != num_moments_t or
-        file_num_groups != num_groups_t or file_num_local_dofs != num_local_dofs or
-        file_num_local_cells != num_local_cells)
-    {
-      std::stringstream outstr;
-      outstr << "num_local_nodes: " << file_num_local_nodes << " vs " << num_local_nodes << "\n";
-      outstr << "num_moments_    : " << file_num_moments << " vs " << num_moments_t << "\n";
-      outstr << "num_groups     : " << file_num_groups << " vs " << num_groups_ << "\n";
-      outstr << "num_local_dofs : " << file_num_local_dofs << " vs " << num_local_dofs << "\n";
-      outstr << "num_local_cells: " << file_num_local_cells << " vs " << num_local_cells << "\n";
-      Chi::log.LogAll() << "Incompatible DOF data found in file " << file_name << "\n"
-                        << "File data vs system:\n"
-                        << outstr.str();
-      file.close();
-      return;
-    }
+  // Check compatibility with system macro info
+  const auto uk_man = flux_moments_uk_man_;
+  const auto NODES_ONLY = UnknownManager::GetUnitaryUnknownManager();
+
+  const uint64_t num_local_cells = grid_ptr_->local_cells.size();
+  const uint64_t num_local_nodes = discretization_->GetNumLocalDOFs(NODES_ONLY);
+  const auto num_local_dofs = discretization_->GetNumLocalDOFs(uk_man);
+
+  // clang-format off
+  ChiLogicalErrorIf(file_num_local_cells != num_local_cells,
```

Comment:
@andrsd is there an upcoming PR with more name changes, for things like these?

Reply:
Yes, there will be a PR (or to be more precise PRs(!)) for things like this.